runtime/v2/shim/shim.go

```diff
@@ -396,6 +406,10 @@ func serve(ctx context.Context, server *ttrpc.Server, signals chan os.Signal) er
 			dumpStacks(logger)
 		}
 	}()
+
+	// We handle exit signals separately because we need to run things and do not want to block the reaper from doing its work.
+	go handleExitSignals(ctx, logger, shutdown)
```

Comment:
We can... what if we renamed `handleSignals` to `reap`?
I don't really like having the shutdown happen in the same loop, even if we do spin up a goroutine because we can't exit out of the loop safely.

Reply:
I am also not a fan of boundless goroutines.

Reply:
> what if we renamed handleSignals to reap?

Sure! The rename sounds good and reasonable to me.

Reply:
Updated.